adapters/repos/db/vector/hnsw/snapshot.go

```diff
@@ -0,0 +1,108 @@
+package hnsw
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// PauseMaintenance makes sure that no new background processes can be started.
+// If a Combining or Condensing operation is already ongoing, the method blocks
+// until the operation has either finished or the context expired
+//
+// If a Delete-Cleanup Cycle is running (TombstoneCleanupCycle), it is aborted,
+// as it's not feasible to wait for such a cycle to complete, as it can take hours.
+func (h *hnsw) PauseMaintenance(ctx context.Context) error {
+	cleanupHalted := make(chan struct{})
+
+	go func() {
+		h.commitLog.PauseMaintenance()
```

Comment:
as I understand PauseMaintenance just sends signal to SwitchLogs and CombineAndCondenseLogs to stop, but does not wait for theirs result, so we are not sure if it was stopped or not?

we also do not know if both were successfully stopped, just one or none in given timeout. so we do not know which one to resume later on

Reply:
Fixed